index.js

```diff
@@ -21,72 +37,98 @@ var resultStrings = [
   'MISMATCH'
 ];
 
-exports.resultCodeToString = function(code) {
-  if (code < 1 || code >= resultStrings.length) { return; }
-  return resultStrings[code];
+HmacAuth.resultCodeToString = function(code) {
+  return resultStrings[code - 1];
 };
 
 function signedHeaders(req, headers) {
-  return headers.map(function(header) { return req.get(header) || ''; });
+  return headers.map(function(header) {
+    var value = req.headers[header];
+    if (typeof value === Array) { value = value.join(','); }
```

Comment:
Actually, that would miss `undefined` values, and since this is specifically for Node and we don't need to worry about older browsers, how about using `if (Array.isArray(value) { ... }`?